docs/reference/filters.md

```diff
@@ -1848,6 +1848,79 @@ Path("/cheap")     -> clusterLeakyBucketRatelimit("user-${request.cookie.Authori
 Path("/expensive") -> clusterLeakyBucketRatelimit("user-${request.cookie.Authorization}", 1, "1s", 5, 2) -> ...
 ```
 
+## shedder
+
+The basic idea of load shedding is to reduce errors by early stopping
+partial of the ingress requests that create too much load and serve
+the maximum throughput the system can do.
+
+There is a great talk by [Acacio Cruz from
+Google](https://www.youtube.com/watch?v=XNEIkivvaV4&feature=youtu.be)
+that explains the basic principles.
+
+### admissionControl
+
+Implements an admission control filter similar to [envoy admission control](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/admission_control_filter#admission-control).
+
+The probability of rejection is calculated by the following equation:
+
+$$ P_{reject} = ( { n_{total} - { n_{success} \over threshold } \over n_{total} + 1} )^{1 \over aggression } $$
+
+Examples:
+
+    admissionControl(metricSuffix, mode, d, windowSize, minRPS, successThreshold, maxRejectProbability, aggression)
+    admissionControl("myapp", "active", "1s", 5, 10, 0.95, 0.9, 2.0)
+
+Parameters:
+
+* metric suffix (string)
+* mode (enum)
+* d (time.Duration)
+* window size (int)
+* min requests (int)
+* success threshold (float64)
+* max reject probability (float64)
+* aggression (float64)
+
+Metric suffix is the chosen suffix key to expose reject counter,
+should be unique by filter instance
+
+Mode has 3 different possible values and defaults to "inactive":
+
+* "active" will reject traffic
+* "inactive" will never reject traffic (default)
+* "log" will not reject traffic, but log to debug filter settings
+
+D the time duration of a single slot for required counters in our
+circular buffer of window size and default to 1s.
```

Comment:
For leaky bucket I decided not to use any defaults to make users understand what they are doing and because there is no one-size-fits all config for load schedding/ratelimit IMO.

Reply:
I want to adjust defaults when we know there might be known good configuration for the bulk of services at best.
I don't think it's reasonable that the users that have no idea about the thing can judge if windowSize should be 5 or 50 and I want to make default a good choice and I tried to sort it such that simple things are in front and complicated things are last.
So it was a careful choice, that I would like to keep.

Reply:
Agreed on first without defaults and later if we observe known good values, we can add defaults